solidity/contracts/domains/interfaces/IZetoFungible.sol

```diff
@@ -10,6 +10,7 @@ interface IZetoFungible {
     struct TransferParam {
         string to;
         uint256 amount;
+        bytes data;
```

Comment:
I chose to put the `data` inside this struct, so it is "per transfer" rather than "per transaction" - since all Zeto methods are built for batching.

However, because we ultimately just end up with a list of input, output, and info states, it will be difficult to "match" transfers and data when parsing events.

Reply:
I spent some more time thinking about this and considering the alternate option (placing a single `data` outside the list of transfers). Ultimately that option didn't feel better. State distribution was a little fuzzy - in the current model it's clear that the "data" states get distributed to the same recipients as the "coin" states, but with a single "data" state, it has to be distributed to all recipients that received a coin as part of the transfer.

That isn't necessarily wrong, but I couldn't convince myself it was better than what is here now.